mesonbuild/cmake/tracetargets.py

```diff
@@ -45,6 +45,34 @@ def __init__(self) -> None:
         self.public_compile_opts: T.List[str] = []
         self.libraries:           T.List[str] = []
 
+# Get a property that can be overriden depending on the configuration name
+# Example: IMPORTED_LOCATION which can be overriden as IMPORTED_LOCATION_RELEASE
+# https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_CONFIGURATIONS.html#imported-configurations
+def get_config_declined_property(target: CMakeTarget,
+                                 property: str,
+                                 trace: 'CMakeTraceParser') -> T.List[str]:
+    cfg = trace.build_type.upper() if trace.build_type else None
+    imported_cfgs: T.List[str] = []
+    if f'MAP_IMPORTED_CONFIG_{cfg}' in target.properties:
+        cfg = target.properties[f'MAP_IMPORTED_CONFIG_{cfg}'].upper()
+    if 'IMPORTED_CONFIGURATIONS' in target.properties:
+        imported_cfgs = [x.upper() for x in target.properties['IMPORTED_CONFIGURATIONS'] if x]
+        if cfg not in imported_cfgs:
+            if cmake_is_debug(trace.env):
+                if 'DEBUG' in imported_cfgs:
+                    cfg = 'DEBUG'
+                elif 'RELEASE' in imported_cfgs:
+                    cfg = 'RELEASE'
+            else:
+                if 'RELEASE' in imported_cfgs:
+                    cfg = 'RELEASE'
```

Comment:
I think we can simplify this to:
```ptyon
if cmake_is_debug(trace.env) and 'DEBUG' in imported_cfgs:
    cfg = 'DEBUG'
else:
    cfg = 'RELEASE'
```

Reply:
This is the previous author's version, I simplified it